app/assets/javascripts/directives/repository/valid_unc_path.js

```diff
@@ -1,41 +1,36 @@
-ManageIQ.angularApplication.directive('checkpath', ['miqService', function (miqService){
+ManageIQ.angularApplication.directive('checkpath', function() {
   return {
     require: 'ngModel',
-       link: function (scope, elem, attrs, ctrl) {
-         scope.$watch("repoModel.repo_path", function() {
-           if(ctrl.$modelValue != undefined && scope.afterGet) {
-             setValidity(scope, ctrl, ctrl.$modelValue, false);
-           }
-         });
-         ctrl.$parsers.push(function(value) {
-           setValidity(scope, ctrl, ctrl.$viewValue, true);
-           return value;
-          });
+    link: function (scope, elem, attr, ctrl) {
 
-      validPath = function(scope, path, bClearMsg) {
+      scope.$watch(attr.ngModel, function() {
+        if((ctrl.$modelValue != undefined && scope.afterGet) || scope.formId == "new") {
+            setValidity(scope, ctrl.$modelValue);
+        }
+      });
+
+      ctrl.$parsers.push(function(value) {
```

Comment:
Hmm, we may want to move the validator functionality from `$parsers` to `$validators`, that was just a hack for angular <1.3, right?